src/map/packet_system.cpp

```diff
@@ -1054,6 +1054,8 @@ void SmallPacket0x01A(map_session_data_t* const PSession, CCharEntity* const PCh
             PChar->animation = ANIMATION_NONE;
             PChar->updatemask |= UPDATE_HP;
             PChar->StatusEffectContainer->DelStatusEffectSilent(EFFECT_MOUNTED);
+            // Workaround for a bug where dismounting out of update range would cause the character to stop rendering.
+            PChar->loc.zone->PushPacket(PChar, CHAR_INZONE, new CCharPacket(PChar, ENTITY_UPDATE, UPDATE_HP));
```

Comment:
I think there must be a problem with the update mask in packet we send out when chars come into range. The reason your fix works is it's sending it to the entire zone regardless, but those players should receive that data when the char becomes within 50 yalm (we intentionally don't send out entity updates for things beyond that distance). If it works it works, better this than invisible players, but we'll have to figure the rest out later.

Reply:
> This fix has a side effect of sending the associated dismount packet to all characters in the zone. I consider this low impact, but notable enough that I'd consider this fix temporary. A better solution would be to figure out where the packets the server is sending differ from retail.

Oh you already said as much. Well that will learn me to read code before the description. I shall now resume my work at the department of redundancy department.

Reply:
I tested this and tried force setting the update mask to always include UPDATE_HP (in Windower it's called Update Vitals). The 0x00D packet with the flag forcefully set would get received, but the client would just ignore it.

Interestingly, this bug doesn't affect chocobo mounts. Only non-chocobo mounts. This leads me to believe it's not the update mask, but rather there's some element of the packet that is not constructed how the client expects. Either that, or the client requires the packet to include *only* UPDATE_HP and not also UPDATE_POSITION in order to properly register the dismount, despite the combined update working for other cases.

In any case, I don't know anyone on retail so I can't test of the packet itself is part of the problem.

Reply:
If you describe what you need I can grab caps for you. This is rate-limited by the client, right? It isn't possible to hop on and off a mount any more than once every 30s? So it isn't really at risk of hurting performance

Reply:
> If you describe what you need I can grab caps for you. This is rate-limited by the client, right? It isn't possible to hop on and off a mount any more than once every 30s? So it isn't really at risk of hurting performance

That would be great. To capture, you'll need a friend/second account with a non-Chocobo mount. I want the captures from the observing/passive character, *not* the mounting/dismount character.

I'd like five captures, all similar with minor variations: dismounting out of range with with a non-chocobo mount and with a chocobo, the same band finally without using a mount.

1. Have two characters. We'll call the mounting character M, and the capturing character C. They should both be in a zone where mounting is allowed.
2. M mounts a non-Chocobo mount and gets within 45 yalms of C, such that C and M can see eachother.
3. M walks at least 50 calms away from C, such that neither M nor C can see eachother.
4. M

Reply:
I also posted the capture request in the FFXI Captures discord.

Reply:
@zach2good